content/en/blog/_posts/2025-04-09-ensure-secret-pulled-images/index.md

```diff
@@ -0,0 +1,145 @@
+scale up, or pod restart), the image registry that provides the image MUST be available, making
+the image registry critical for stability of services running inside of the cluster.
+
+## How it all works
+
+The feature is based on persistent, file-based caches that are present on each of
+the nodes. The following is a simplified description of how the feature works,
+for the complete version, please see [KEP-2535](https://github.com/kubernetes/enhancements/blob/0262549808ebe7e64b94ee6f8e4fbdd024332470/keps/sig-node/2535-ensure-secret-pulled-images/README.md).
+
+The process of requesting an image for the first time goes likes this:
+  1. A pod requesting an image from a private registry is scheduled to a node.
+  1. The image is not present on the node and so the Kubelet makes a record about
+     the intention to pull the image.
+  1. The Kubelet extracts credentials from the Kubernetes Secret referenced by the Pod
+     as an image pull secret, and uses them to pull the image from the private registry.
+  1. After the image has been successfully pulled, the Kubelet makes a record about
+     the successful pull, including the details about credentials used and the Secret
+     they originated from.
+  1. The Kubelet removes the original record about the intent to pull an image.
+
+When future pods scheduled to the same node request the same private image that was pulled before:
+  1. The Kubelet checks the credentials that the pod provided for the pull.
+  1. If the credential hash, or the source Secret of the credentials match the hash or source
+     Secret which were recorded for a successful pull before, the pod is allowed to use
+     the already-pulled image.
+  1. If the credentials or their source Secret are not found in the records of
+     previous successful pulls for that image, the Kubelet will attempt to use
+     these new credentials to request a pull from the remote registry.
+
+## Try it out
+
+In Kubernetes 1.33 we shipped the alpha version of this feature. To give it a spin,
+enable the `KubeletEnsureSecretPulledImages` feature gate for your 1.33 Kubelets.
+
+You can learn more about the feature and additional optional configuration on the
+[concept page for Images](/docs/concepts/containers/images/#ensureimagepullcredentialverification)
+in the official Kubernetes documentation.
+
+## What's next?
+
+In future releases, there are a couple of areas for us to focus on:
+1. Make the feature work together with [Projected service account tokens for Kubelet image credential providers](https://kep.k8s.io/4412) which
+   adds a new, workload-specific source of image pull credentials.
+1. Write a benchmarking suite to be able to measure the performance of the feature,
+   and to be able to measure the performance impact of any future changes to it.
+1. Implement an in-memory caching layer so that we don't need to read files for each image pull request.
```

Comment:
```suggestion
1. Implement an in-memory caching layer so that we don't need to read files for each image pull request.
1. Add support for credential expirations, thus forcing previously validated credentials to be re-authenticated.  
```

Reply:
👍